src/backend/mod.rs

```diff
@@ -85,4 +85,32 @@ impl DNSBackend {
 
         DNSResult::Success(results)
     }
+
+    // reverse lookup must return a single name resolved via mapping
+    pub fn reverse_lookup(&self, requester: &IpAddr, lookup_ip: &str) -> String {
```

Comment:
@mheon I am fine to do it but it would only make difference if we have like `100000`+ of entries, the amount of entries we have hardly makes any difference now.

Would it be possible if we could add another table later if we ever hit bottlenecks ? Cause as of now i think it should not make any difference.